src/Common/AzureRest/IAzureRestOperations.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Microsoft.Rest.Azure;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.WindowsAzure.Commands.Common.AzureRest
+{
+    /// <summary>
+    /// AzureRest operations.
+    /// </summary>
+    public partial interface IAzureRestOperations
+    {
+        /// <summary>
+        /// Support universal HTTP request
+        /// </summary>
+        /// <typeparam name="T">Type of return object</typeparam>
+        /// <param name="method">Http Method</param>
+        /// <param name="path">The path compoment in URL</param>
+        /// <param name="queries">The queries compoment in URL</param>
+        /// <param name="fragment">The fragment compoment in URL</param>
+        /// <param name="content">The content in request body</param>
+        /// <param name="customHeaders">The headers that will be added to request.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <exception cref="Microsoft.Rest.Azure.CloudException">
+        /// Thrown when the operation returned an invalid status code
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null
+        /// </exception>
+        Task<AzureOperationResponse<T>> BeginHttpMessagesAsync<T>(HttpMethod method, string path, Dictionary<string, List<string>> queries = null, string fragment = null, string content = null, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default);
```

Comment:
recommend to use interface instead of class for parameters: e.g. IDictionary<>, IList<>